script/src/verify/tests/ckb_latest/features_since_v2021.rs

```diff
@@ -685,3 +686,116 @@ fn check_typical_secp256k1_blake160_2_in_2_out_tx_with_complete() {
+
+    let (dyn_lib_cell, dyn_lib_data_hash) = {
+        let dyn_lib_cell_data = Bytes::from(
+            std::fs::read(
+                Path::new(env!("CARGO_MANIFEST_DIR"))
+                    .join("testdata/dyn_load_code/dyn_load_code_lib"),
+            )
+            .unwrap(),
+        );
+        let dyn_lib_cell_output = CellOutput::new_builder()
+            .capacity(Capacity::bytes(dyn_lib_cell_data.len()).unwrap().pack())
+            .build();
+        let dyn_lib_data_hash = blake2b_256(&dyn_lib_cell_data);
+        let dyn_lib_cell =
+            CellMetaBuilder::from_cell_output(dyn_lib_cell_output, dyn_lib_cell_data)
+                .transaction_info(default_transaction_info())
+                .build();
+        (dyn_lib_cell, dyn_lib_data_hash)
+    };
+
+    let rtx = {
+        let args: packed::Bytes = {
+            let number = 0x01u64; // a random odd value
+
+            let mut vec = Vec::with_capacity(8 + dyn_lib_data_hash.len());
+            vec.extend_from_slice(&number.to_le_bytes());
+            vec.extend_from_slice(&dyn_lib_data_hash);
+            vec.pack()
+        };
+
+        let dyn_lock_cell_data = Bytes::from(
+            std::fs::read(
+                Path::new(env!("CARGO_MANIFEST_DIR"))
+                    .join("testdata/dyn_load_code/dyn_load_code_lock"),
+            )
+            .unwrap(),
+        );
+        let dyn_lock_cell_output = CellOutput::new_builder()
+            .capacity(Capacity::bytes(dyn_lock_cell_data.len()).unwrap().pack())
+            .build();
+        let dyn_lock_script = Script::new_builder()
+            .hash_type(script_version.data_hash_type().into())
+            .code_hash(CellOutput::calc_data_hash(&dyn_lock_cell_data))
+            .args(args)
+            .build();
+        let output = CellOutputBuilder::default()
+            .capacity(capacity_bytes!(100).pack())
+            .lock(dyn_lock_script)
+            .build();
+        let input = CellInput::new(OutPoint::null(), 0);
+
+        let transaction = TransactionBuilder::default().input(input).build();
+
+        let dummy_cell = CellMetaBuilder::from_cell_output(output, Bytes::new())
+            .transaction_info(default_transaction_info())
+            .build();
+        let dyn_lock_cell =
+            CellMetaBuilder::from_cell_output(dyn_lock_cell_output, dyn_lock_cell_data)
+                .transaction_info(default_transaction_info())
+                .build();
+
+        ResolvedTransaction {
+            transaction,
+            resolved_cell_deps: vec![dyn_lock_cell, dyn_lib_cell],
+            resolved_inputs: vec![dummy_cell],
+            resolved_dep_groups: vec![],
+        }
+    };
+
+    let mut cycles = 0;
+    let cycles_step_1 = 100_000;
+    let max_cycles = Cycle::MAX;
+    let verifier = TransactionScriptsVerifierWithEnv::new();
+    // TODO fix resume from snapshot
+    let should_be_invalid_permission = script_version <= ScriptVersion::V1;
```

Comment:
Change to the follow code after fixing the snapshot:

```suggestion
    let should_be_invalid_permission = script_version < ScriptVersion::V1;
```